mesop/server/config.py

```diff
@@ -24,6 +24,8 @@ class Config(BaseModel):
   state_session_backend_firestore_collection: str = "mesop_state_sessions"
   state_session_backend_sql_connection_uri: str = ""
   state_session_backend_sql_table: str = "mesop_state_sessions"
+  static_folder: str = ""
```

Comment:
I think we need to go with B for now just for the edge case of someone coincidentally already having a static folder.

I agree that we will need the ability to expose multiple static folders. That would become more important for bigger Mesop apps. I tried a few different interfaces for exposing multiple static folders, but ultimately wasn't sure, so decided to start with just one static folder. And I'm hoping that there will still be a path to expand to multiple static folders in the future

I think for multiple static folders, it would become more of a necessity if there were a pattern where Mesop could be used more modularly. Than those shared components may need their own static folder. In Flask, this would be akin to the Flask Blueprints.

In terms of the web component use case, I think that those shouldn't need to go into the same static folder.

I think would just need to update the expected format for web components.

So for this route, I think we should do something like:

```
 @app.route(f"/{WEB_COMPONENTS_PATH_SEGMENT}/static/<path:path>")
 def serve_web_components(path: str):
```

I think then the python files and javascript/css files can exist in the same folder and be more easily packageble. But the web viewable output would be in the static folder.

This would be a breaking change though.

Agree about updating the docs. Just wanting to make sure we settled on the interface for this first.

Reply:
OK sounds good.

Not sure if I totally understand the breaking change that you're proposing - is it that web components have to be in a separate directory from the Python code? Maybe it's just me, but I do like having the Python and JS files right next to each other (in the same dir) because they are so closely linked.

We could offer both though:

```py
@mel.web_component(path="./inner_component.js", static_folder="./static/")
```

This way you can keep the entry point file in the same directory (or somewhere else) and optionally define a static folder for JS modules / other things that are imported by the entry point.

Taking a step back though, I'm not super optimistic about shared web components (at least in the short-term) so I I'm not too worried about us solving this soon. On one hand there's the performance / shared library issue (see above) which isn't really easy to solve unless you vendor in web components (i.e. copy paste) and on the other hand, it doesn't seem like anyone in the open-source community is really building non-trivial web components that you would want to share (vs. just vendoring)

Reply:
Yeah I don't have a strong stance on this. So I think having a static folder option could be a nice compromise for people who prefer having the files in the same directory. I think with the static folder, then one could include images or other non-css / js assets a bit easier. Otherwise from what I understand, we'd have to do some kind of allowlisting of the files in the folder that are accessible.